pkg/controller/source/source.go

```diff
@@ -31,6 +32,11 @@ import (
 	logf "github.com/kubernetes-sigs/controller-runtime/pkg/runtime/log"
 )
 
+const (
+	// initialBufferSize is the initial number of event notifications that can be buffered.
+	initialBufferSize = 1024
```

Comment:
lets make this a variable on the ChannelSource and write a test for what happens when we run out of buffer (e.g. we can set the buffer size to 1)

Reply:
So when do you think it should be specified?
Or we'll in fact always use the default value, and the field will only be changed in test?